### DISCUSSION TITLE
Shell Command "6top help" unavailable

### DISCUSSION THREAD

**Discussion by mehdisfh** [contiki-ng/contiki-ng#2441] [Q&A] [answered]
Hi everyone , 

I added 6top module to my Makefile , but its shell command "**6top help**" doesn't work. can someone tell me what should i do ? should i create this shell command by myself? I also tried "**6top-help**" but it still doesn't work. i get always get this error :  **command is unavailable** 

thnks in advance

**Comment by bartvdbraak** [comment, marked as answer]
Hello @mehdisfh,

It seems that you're experiencing issues with the 6top module in Contiki-NG. To address the problem with the "6top help" shell command, here are a few suggestions:

1. Verify Module Installation:
   - Ensure that the 6top module is correctly installed and integrated into your Contiki-NG project.
   - Double-check your Makefile to confirm that you have added the module to the appropriate sections (e.g., MODULES, APPS, etc.).
   - Make sure that the module's source code files are present in the correct directory within your project structure.

2. Confirm Shell Command:
   - Review the module's documentation, if available, to verify the correct syntax for accessing the help command. It's possible that the command may have a slightly different format.
   - Check the official Contiki-NG documentation or the module's specific documentation for any insights into the proper usage of the "6top help" command.

3. Custom Shell Command:
   - If you cannot find an existing "help" command for the 6top module, it might be necessary to create one yourself.
   - Examine the module's source code and any provided examples to see if there are guidelines or instructions for implementing custom shell commands.
   - Alternatively, you can try reaching out to the Contiki-NG community or the module's maintainers for assistance on how to create the "help" command for the 6top module.

4. Command Availability:
   - The "command is unavailable" error suggests that either the command is not recognized by the shell or it may not have been properly implemented or registered within the Contiki-NG framework.
   - Ensure that the module's functionality, including the "help" command, is fully supported and integrated into the version of Contiki-NG you are using. It's possible that the feature might be available in a different Contiki-NG release.

If you require further assistance, I recommend providing additional details about your project setup, the version of Contiki-NG you are using, and any specific steps you have taken so far. This will help community members better understand the issue and provide more targeted guidance.

Best of luck in resolving the problem with the 6top module!

**Reply by mehdisfh** [reply]
Dear @bartvdbraak , 

Many Thanks for your valuable Answer , I'll follow your Tips and hope to solve the problem.

**Reply by bartvdbraak** [reply]
Thank you and good luck @mehdisfh!

If my reply was useful to you, please consider **marking it as answer** to your question:

![image](https://github.com/contiki-ng/contiki-ng/assets/3996360/fa867628-7ffe-43d1-a2de-96b2463c510c)

**Reply by mehdisfh** [reply]
I'll do , Thanks

**Reply by bartvdbraak** [reply]
Thank you, this discussion is currently still marked as **Unanswered** 😉.
